controllers/front/ProductController.php

```diff
@@ -1319,13 +1319,29 @@ protected function getRequiredQuantity($product)
         return $requiredQuantity;
     }
 
-    public function getBreadcrumbLinks()
+    /**
+     * Generates breadcrumb according to product categories and his category subtree.
+     * If the product is accessed from another category than product default category, it will generate the breadcrumb according to current category.
+     *
+     * @return array
+     *
+     * @throws PrestaShopException
+     */
+    public function getBreadcrumbLinks(): array
     {
         $breadcrumb = parent::getBreadcrumbLinks();
 
-        $categoryDefault = new Category($this->product->id_category_default, $this->context->language->id);
+        // $productBreadcrumbCategory can have two possible values
+        // - current : Category the product was accessed from
+        // - default : Product default category
+        if (('current' === Configuration::get('PS_PRODUCT_BREADCRUMB_CATEGORY')) &&
+            !is_null($this->category)) {
```

Comment:
```suggestion
        if (('current' === Configuration::get('PS_PRODUCT_BREADCRUMB_CATEGORY'))) {
```

Is there any scenario when category could be null? 🤔

Reply:
Product not assigned to any category, or it's category was deleted and it wasn't handled properly.